### DISCUSSION TITLE
Get triangles from Instanced asset?

### DISCUSSION THREAD

**Discussion by redstonedash** [google/filament#4339] [Q&A]
Exactly as the title says. I have a instanced asset an I need to get the triangle information off of it so I can provide that to my physics engine, working in an older version of filament but will update if I need to.

**Comment by pixelflinger** [comment]
This is not supported and won't be supported.
Filament actually doesn't keep triangle information, it's given to the underlying backend (e.g. opengl) and no cpu copy is kept around.
The way to do this, is for the application to keep the geometry it needs for physics, which in most cases will be different (simpler) anyways.

**Reply by jnmaloney** [reply]
Would it be possible to support callbacks on load, similar to the way filamesh loader does?

**Reply by redstonedash** [reply]
Well. How about a way to get triangle information from a asset loader?

**Reply by prideout** [reply]
As pixelflinger said, this isn't something we would support in Filament.  Maybe you could grab the data in your loader.  For example if you are using our gltfio loader library then you can access the raw cgltf structs via `getSourceAsset()`.
